### PR TITLE
Add flags for adaptive mesh refinement

### PR DESCRIPTION
**Pull request by kidder**
### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — COMMENTED** [review]
@nilsdeppe please look at

**Review by wthrowe — DISMISSED** [review]

**Review by nilsdeppe — DISMISSED** [review]

File: src/Domain/Amr/Flag.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines enum class Side.
```
Comment (nilsdeppe) [inline src/Domain/Amr/Flag.hpp]: But fix this :)

Comment (kidder) [reply src/Domain/Amr/Flag.hpp]: oops!

**Review by nilsdeppe — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe I squashed a typo fixup

**Review by wthrowe — APPROVED** [review]

**Comment by kidder** [comment]
code coverage failure is from unreachable ERROR

**Pull request merged**